app/services/cfe/regular_transactions_payload_service.rb

```diff
@@ -1,6 +1,7 @@
 module Cfe
   class RegularTransactionsPayloadService < BaseService
     def call
+      return if early_eligibility?
```

Comment:
I'm guessing you've added in this guard clause in because this method blows up otherwise? If so, I think this highlights an important problem: Cfe methods are designed to blow up if presented with an incomplete or invalid session, and the early eligibility result depends on calling Cfe with an invalid session. This is quite hard to unpick, e.g. you _can't_ just exit early out of this method because it's in this method that we add a bunch of gross income  data. This method packages up a bunch of income and outgoings data into the same place.

I think the general approach (pass in an 'early eligibility' flag, and when it's set to true, get Cfe to work with what it's got instead of blowing up), is probably the right one, but sadly I think it's going to be a case of doing more than guard clauses. We're going to have to go all the way into, e.g. CfeParamBuilders::RegularTransactions so that if needed it can add data from the income_form but not the outgoings_form.

Reply:
Thanks for this! I added this is as a first step to stop things blowing up as you've said, to sort of get on the right track - will think on this one based on your comment